lib/ClusterShell/NodeUtils.py

```diff
@@ -224,12 +227,12 @@ def _upcall_cache(self, upcall, cache, key, **args):
             raise GroupSourceNoUpcall(upcall, self)
 
         # Purge expired data from cache
-        if key in cache and cache[key][1] < time.time():
+        if key in cache and cache[key]:
```

Comment:
That doesn't look right, you're purging the whole cache instead of a time-based expiration?

Reply:
Agree.  Needs to be fixed.  Was seeing a problem where `cache[key][1]` was not defined and was blowing up.  Did this temporarily and forgot to go back and fix.  Thanks!